pycytominer/cyto_utils/collate.py

```diff
@@ -0,0 +1,242 @@
+
+    Parameters
+    ----------
+    batch : str
+        Batch name to process
+    config : str
+        Config file to pass to cytominer-database
+    plate : str
+        Plate name to process
+    base_directory : str, default "../.."
+        Base directory where the CSV files will be located
+    column : str, optional, default None
+        An existing column to be explicitly copied to a Metadata_Plate column if Metadata_Plate was not set
+    munge : bool, default False
+        Whether munge should be passed to cytominer-database, if True will break a single object CSV down by objects
+    pipeline : str, default 'analysis'
+        A string used in path creation
+    remote : str, optional, default None
+        A remote AWS directory, if set CSV files will be synced down from at the beginning and to which SQLite files will be synced up at the end of the run
+    aggregate_only : bool, default False
+        Whether to perform only the aggregation of existent SQLite files and bypass previous collation steps
+    tmp: str, default '/tmp'
+        The temporary directory to be used by cytominer-databases for output
+    overwrite: bool, optional, default False
+        Whether or not to overwrite an sqlite that exists in the temporary directory if it already exists
+    add_image_features: bool, optional, default True
+        Whether or not to add the image features to the profiles
+    image_feature_categories: list, optional, default ['Number','Intensity','Granularity','Texture','ImageQuality','Count','Threshold']
+        The list of image feature groups to be used by add_image_features during aggregation
+    printtoscreen: bool, optional, default True
+        Whether or not to print output to the terminal
+    """
+
+    from pycytominer.cyto_utils.cells import SingleCells
+
+    # Set up directories (these need to be abspaths to keep from confusing makedirs later)
+    input_dir = os.path.abspath(
+        os.path.join(base_directory, "analysis", batch, plate, pipeline)
+    )
+    backend_dir = os.path.abspath(os.path.join(base_directory, "backend", batch, plate))
+    cache_backend_dir = os.path.abspath(os.path.join(temp, "backend", batch, plate))
+
+    aggregated_file = os.path.join(backend_dir, plate + ".csv")
+    backend_file = os.path.join(backend_dir, plate + ".sqlite")
+    cache_backend_file = os.path.join(cache_backend_dir, plate + ".sqlite")
+
+    if not aggregate_only:
+        if os.path.exists(cache_backend_file):
+            if not overwrite:
+                sys.exit(
+                    f"An SQLite file for {plate} already exists at {cache_backend_file} and overwrite is set to False. Terminating."
+                )
+            else:
+                os.remove(cache_backend_file)
+
+        for eachdir in [input_dir, backend_dir, cache_backend_dir]:
+            if not os.path.exists(eachdir):
+                os.makedirs(eachdir, exist_ok=True)
+
+        if remote:
```

Comment:
I'm not a huge fan of hardcoding support for AWS, but given the band-aid nature of adding cytominer-database support to pycytominer, I'm ok with it as a short-term patch